### PR TITLE
View and alter Order APIs & SQL Enums update

### PR DESCRIPTION
**Pull request by SteveGuoh**
- Created get orders API and filter orders by order status and order type API, all tested via PostMan

### CODE REVIEW

File: src/main/java/com/backend/OrderHere/mapper/Order/OrderMapper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.backend.OrderHere.model.User;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+
+@Component
+public class OrderMapper {
+
+    public OrderGetDTO fromOrderToOrderGetDTO(Order order) {
```
Comment (Zachary1319) [inline src/main/java/com/backend/OrderHere/mapper/Order/OrderMapper.java:14]: Suggest we still also use `mapstruct` to do the mapping.
Some code like:
```
@Mapper(componentModel = "spring")
public interface OrderMapper {
    @Mapping(source = "user.userId", target = "userId")
    @Mapping(source = "user.username", target = "userName")
    OrderGetDTO fromOrderToOrderGetDTO(Order order);
}
```

File: src/main/resources/db/migration/V2__enum_update.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+alter table orders
```
Comment (Zachary1319) [inline src/main/resources/db/migration/V2__enum_update.sql:1]: Feel like the JPA knows how to deal with enum type in DB by `@Enumerated` annotation, do we need to make this change?

Comment (Zachary1319) [reply src/main/resources/db/migration/V2__enum_update.sql:1]: Find a solution to solve the Mapping between Java entity and Postgres enum.
With an external library: https://github.com/vladmihalcea/hypersistence-utils
1.We need to add it as a dependency
`implementation 'io.hypersistence:hypersistence-utils-hibernate-62:3.6.0'`
2.Import it and add annotation to entity like:
```
import io.hypersistence.utils.hibernate.type.basic.PostgreSQLEnumType;
...
@Enumerated(EnumType.STRING)
@Column(name = "order_status", nullable = false, columnDefinition = "order_status")
@Type(PostgreSQLEnumType.class)
private OrderStatus orderStatus;

@Enumerated(EnumType.STRING)
@Column(name = "order_type", nullable = false, columnDefinition = "order_type")
@Type(PostgreSQLEnumType.class)
private OrderType orderType;
```

reference: 
https://vladmihalcea.com/the-best-way-to-map-an-enum-type-with-jpa-and-hibernate/
https://stackoverflow.com/questions/76567760/hibernate-6-2-mapping-between-java-enums-and-postgres-enums

Tested some APIs:
image
image

Comment (SteveGuoh) [reply src/main/resources/db/migration/V2__enum_update.sql]: 👍🏻

**Pull request merged**